src/main/java/com/hivemq/bootstrap/netty/ChannelDependencies.java

```diff
@@ -281,4 +285,8 @@ public InterceptorHandler getInterceptorHandler() {
     public GlobalMQTTMessageCounter getGlobalMQTTMessageCounter() {
         return globalMQTTMessageCounter;
     }
+
+    public @NotNull PublishSendHandler getPublishSendHandler() {
+        return new PublishSendHandler(metricsHolder);
```

Comment:
why was the constructor call moved here? Imho it was better placed in the channel initializer.
The getter pretends that you get the same instance when invoked multiple times.

Reply:
Seems feasible. Idk if @DC2-DanielKrueger had any special reason for it.

Reply:
well, all deps are here and u got the metricsholder there. I would rename the method to createPublishSendHandler(); get is wrong semenantic

Reply:
I agree with Daniel, that `ChannelDependencies` should be the place for providing all the Handlers as a single point of retrieval. Still, I am unsure about the naming even with `create` as all the other methods are Getters or are using a Provider...